Kernel/Net/Routing.cpp

```diff
@@ -129,6 +130,24 @@ void update_arp_table(IPv4Address const& ip_addr, MACAddress const& addr, Update
     }
 }
 
+SpinlockProtected<Route::RouteList>& routing_table()
+{
+    return *s_routing_table;
+}
+
+void update_routing_table(IPv4Address const& destination, IPv4Address const& gateway, IPv4Address const& netmask, RefPtr<NetworkAdapter> const adapter, UpdateTable update)
+{
+    auto route_entry = adopt_ref_if_nonnull(new (nothrow) Route { destination, gateway, netmask, adapter });
+    if (!route_entry)
+        return;
+
+    routing_table().with([&](auto& table) {
+        // TODO: Add support for deleting routing entries
+        if (update == UpdateTable::Set)
+            table.append(*route_entry);
```

Comment:
Do we need to check if a route with the same data already exists? 
Seems like an easy way to cause the kernel to allocate memory it doesn't know how to free.